runtime/compiler/makefile.ftl

```diff
@@ -87,6 +87,13 @@ export BUILD_CONFIG?=prod
 <#elseif uma.spec.id?starts_with("linux_arm_linaro")>
   export PLATFORM=arm-linux-gcc-cross
 </#if>
+<#if uma.spec.id?starts_with("osx_x86-64")>
+  <#if uma.buildinfo.version.major == 8>
```

Comment:
@jdekonin this syntax is incorrect. the CI builds are failing because they are unable to generate `runtime/compiler/makefile`:

```
==> if-else  [on line 91, column 3 in compiler/makefile.ftl]

11:45:47 Java backtrace for programmers:
11:45:47 ----------
11:45:47 freemarker.template.TemplateException: The only legal comparisons are between two numbers, two strings, or two dates.
11:45:47 Left  hand operand is a freemarker.template.SimpleScalar
11:45:47 Right hand operand is a freemarker.template.SimpleNumber
11:45:47 
11:45:47 	at freemarker.core.ComparisonExpression.isTrue(ComparisonExpression.java:182)
11:45:47 	at freemarker.core.IfBlock.accept(IfBlock.java:80)
11:45:47 	at freemarker.core.Environment.visit(Environment.java:196)
11:45:47 	at freemarker.core.ConditionalBlock.accept(ConditionalBlock.java:79)
11:45:47 	at freemarker.core.Environment.visit(Environment.java:196)
11:45:47 	at freemarker.core.MixedContent.accept(MixedContent.java:92)
11:45:47 	at freemarker.core.Environment.visit(Environment.java:196)
11:45:47 	at freemarker.core.Environment.process(Environment.java:176)
11:45:47 	at freemarker.template.Template.process(Template.java:232)
11:45:47 	at com.ibm.uma.UMA.generate(UMA.java:215)
11:45:47 	at com.ibm.j9.uma.Main.main(Main.java:277)
11:45:47 
11:45:47 freemarker.template.TemplateException: The only legal comparisons are between two numbers, two strings, or two dates.
11:45:47 Left  hand operand is a freemarker.template.SimpleScalar
11:45:47 Right hand operand is a freemarker.template.SimpleNumber
```

Also, `uma.buildinfo.version.major` is not set. This check won't work even if we fix the syntax.

The correct way is the following:
```
<#if uma.spec.id?starts_with("osx_x86-64")>
ifeq ($(VERSION_MAJOR),8)
  export PLATFORM=amd64-osx-gcc
else
  export PLATFORM=amd64-osx-clang
endif
</#if>
```

Reply:
Thanks @babsingh, I was planning on asking you to test that out.